### ISSUE TITLE
Scan - notifications@expensify.com only changes to Expensify after opening the message in thread

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32581]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.8-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Log in with a new account
2. Go to any chat
3. Create a Scan request using an image that will fail the scanning
4. After the scanning fails, click on the IOU preview twice to open IOU details page
5. Right click on the "scanning failed" system message > Reply in thread

## Expected Result:
In Step 4, the owner of "scanning failed" system message is displayed as Expensify

## Actual Result:
In Step 4, the owner of "scanning failed" system message is [notifications@expensify.com.](mailto:notifications@expensify.com.)
It only changes to Expensify after opening the system message in thread

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/67410c2b-baee-47b4-9f79-16ed8c8acaa6

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
notifications@expensify.com only changes to Expensify after opening the message in thread
### What is the root cause of that problem?
When the account id doesn't exist `getDisplayNameForParticipant` will return the login/email if it can't find the account in `ONYXKEYS.PERSONAL_DETAILS_LIST` so the email is displayed but after opening in thread it will be included in `ONYXKEYS.PERSONAL_DETAILS_LIST` and Expensify will be returned as the display name and hence `Expensify` is displayed
### What changes do you think we should make in order to solve the problem?

We make an exception for `CONST.EMAIL.NOTIFICATIONS` and return `Expensify` as displayName as we did for concierge
```
if (Number(accountID) === CONST.ACCOUNT_ID.NOTIFICATIONS) {
        return {
            accountID,
            displayName: 'Expensify',
            login: CONST.EMAIL.NOTIFICATIONS,
            avatar: UserUtils.getDefaultAvatar(accountID),
        };
    }
```
### What alternative solutions did you explore? (Optional)

**Comment by anmurali** [comment]
When I scan a receipt to a workspace and it fails, this is what I see
<img width="1083" alt="image" src="https://github.com/Expensify/App/assets/14225673/717e3c7e-4d8d-4978-b432-9b71628036c8">


<img width="1088" alt="image" src="https://github.com/Expensify/App/assets/14225673/b073da4b-f8b4-462a-863b-bc1f312088c5">


<img width="1081" alt="image" src="https://github.com/Expensify/App/assets/14225673/29998cc6-3e19-4d69-a402-e06ad8ccd01c">

This seems pretty clear.

**Comment by FitseTLT** [comment]
>> 1. Log in with a new account

@anmurali This issue occurs when logging in a new account where the `notifications@expensify.com` doesn't exist in personal details list on Onyx.

**Comment by anmurali** [comment]
>This issue occurs when logging in a new account where the notifications@expensify.com doesn't exist in personal details list on Onyx.

@FitseTLT what would cause this to be the case? I am not sure if this is actually possible or is an edge case.

**Comment by FitseTLT** [comment]
@anmurali here it is. It occurred when I signed in with a new account and created a request with receipt.
![image](https://github.com/Expensify/App/assets/38649957/e46b5292-82fe-4c73-96f1-58e9e57e6db6)

**Comment by Victor-Nyagudi** [comment]
> 1. Log in with a new account

I can reproduce this on my over 6-month old account with significant amount of activity. 

<details>

<summary>Screenshot of reproduction</summary>

![expensify-notifications-account-display-name-failed-scan](https://github.com/Expensify/App/assets/79470910/a51f2456-44e5-415c-83bf-4038461649cb)

</details>

I believe the one thing my account and a new account have in common is they both don't have the Expensify account ID in `personalDetailsList`, which is why I most likely was able to reproduce the issue. 

@anmurali Based on your earlier [comment](https://github.com/Expensify/App/issues/32581#issuecomment-1852452979), you probably already had the Expensify `accountID` in your `personalDetailsList` from an earlier interaction. 

Here's how I found out the `accountID` was missing. 

When inspecting the props passed to the `ReportActionItemThread` preview below the scan request, I noticed the Expensify avatar isn't shown, and this is because its `accountID` is needed to retrieve the avatar picture from `personalDetailsList`.

<details>

<summary>Screenshot of props</summary>

![expensify-account-ID-prop](https://github.com/Expensify/App/assets/79470910/486bd429-8c19-4357-bf66-a87ab7742315)

</details>

<details>

<summary>Screenshot of missing accountID</summary>

![expensify-missing-account-ID](https://github.com/Expensify/App/assets/79470910/8fbc836f-7809-4e33-aecb-cf9e0b4442a1)

</details>

It's only after I started a reply thread with the Expensify account did the `accountID` get added to `personalDetailsList`, and the avatar and display name got updated.

<details>

<summary>Screenshot of updated personalDetailsList</summary>

![expensify-account-ID-shows-up-after-starting-thread-reply](https://github.com/Expensify/App/assets/79470910/9a5b49a2-3ba1-4ad1-a013-6c7b15249738)

</details>

That being said, I think it would be a good idea to have the Expensify `accountID` added to `personalDetailsList` right after an account is opened given, like Concierge, it also sends automated responses, unless maybe there's some other reason why it's not included there in the first place.

**Comment by anmurali** [comment]
@eVoloshchak thoughts on this bug and @FitseTLT's proposal?

**Comment by eVoloshchak** [comment]
I agree with @Pujan92's comment https://github.com/Expensify/App/issues/32581#issuecomment-1845008701

> How about adding this required entry in the personalDetailsList itself within the backend 'OpenApp' api response, which can avoid more custom checks in front end

While we can fix this on front end, I don't see a reason why `notifications@expensify.com` contact shouldn't be added to personalDetailsList by a back end. This will also ensure we won't have to do the same fix on FE in case other service accounts are added, they can just be updated on the BE.

@anmurali, could you assign an internal engineer to this please?

**Comment by blimpich** [comment]
I meant to get around to looking at this today but didn't get the chance. Will try to take a look at it early next week.

**Comment by anmurali** [comment]
@blimpich - any updates here?

**Comment by blimpich** [comment]
@anmurali not yet, I was OOO monday and then had to take a sick day yesterday. Playing catch up today.

**Comment by blimpich** [comment]
@eVoloshchak can you clarify for me, we're trying to check for / add these personal details whenever the user opens the app? Or just when a new user is created?

**Comment by anmurali** [comment]
@blimpich @eVoloshchak - any updates here?

**Comment by eVoloshchak** [comment]
@blimpich, apologies, missed the original message
> can you clarify for me, we're trying to check for / add these personal details whenever the user opens the app? Or just when a new user is created?

It should be added to personal details whenever an account is created (I assume it would also be there every time user opens the app, unless we clear personal details?)

**Comment by blimpich** [comment]
Sounds good. I don't think we clear personal details, but I also wouldn't be surprised if there is some edge case. Knowing that the main issue though is when creating a new user is helpful. Thank you 🙂

**Comment by blimpich** [comment]
Looked into this a little bit today, its slightly difficult to reproduce on the typical Expensify engineers machine since getting smart scan setup locally is quite a pain. Will keep looking into this.

**Comment by blimpich** [comment]
Banged my head against this today. @Beamanator it looks like you added [this chunk of code](https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.ts#L1583) so that concierge would be handled when we're looking for personal details. Do you remember if you looked into whether this should be done on the backend? All good if you don't remember, 8 months ago is a long time to remember a single block of code and why you put it where it was.

Right now I'm having a hard time figuring out where to put this in the backend. When we request personal details from the backend we do a complicated query to basically say "who does this person know" to see who's information we need, but its based off who the user is interacting with and doesn't really use any constants for Expensify service accounts, so its not clear where exactly this fix should be in the backend. I'll keep looking into this but it is a bit tricky.

**Comment by Beamanator** [comment]
Sorry for the delay @blimpich !

Hmm IIRC (of course i have a perfect memory of previous alex decisions), I added this as a safety net, because we didn't "always" have concierge details available in Onyx - but I like the plan to move this to the backend so we can remove this dumb looking check from the frontend.

Hmm and it sounds like you're looking into some more forward-thinking solution where we can hard-code future expensify account "personal details", right?

My thought would be: Maybe you could add some code near [here](https://github.com/Expensify/Auth/blob/fce038ab55336716fd9e11a88e2be65b86df3b33/auth/command/OpenApp.cpp#L79) which would manually add some "expensify bot accounts" (or whatever we wanna call them) & their personal details to the Onyx response when opening the app?

**Comment by blimpich** [comment]
Update: put out a PR on the backend for fixing this. Waiting on feedback.

**Comment by blimpich** [comment]
Update: PR for backend finally got merged this morning. Waiting for it to deploy. Once its deployed we can check to see if this is still an issue in production.

**Comment by anmurali** [comment]
Adding in BZ backup when I am OOO

**Comment by blimpich** [comment]
Still an issue in production. The personal details list now has notifications@expensify.com in it but it doesn't have a display name so it looks the same as before. Just created a PR to give it a display name, then this should be good.

**Comment by slafortune** [comment]
Thanks for the update @blimpich 🤞

**Comment by blimpich** [comment]
Updated in production. Display name now works. 

https://github.com/Expensify/App/assets/17171674/bdfc0ef7-d749-4c64-a765-574223628c1f

**Comment by blimpich** [comment]
I'm going to do a bit of code cleanup on the frontend, then I will consider this done and close the issue.

**Comment by blimpich** [comment]
Ah I forgot to close this issue. Frontend changes have been merged and a separate issue was created for paying the C+ contributor here: https://github.com/Expensify/App/issues/37190.

Since payment is handled in that issue and this issue has been resolved I'm going to close this now.
